### DISCUSSION TITLE
Result queue approach

### DISCUSSION THREAD

**Discussion by ajfranzoia** [OptimalBits/bull#2729] [Q&A]
I've searched issues/discussions for something related to this but I haven't found anything.

I have a use case where there is a job worker that handles jobs of a single type coming from different producers, located in different services. I am trying to use a results queue approach to handle the result but I haven't found a way to have the proper consumer handle the result. If the worker posts the result as a job to a global queue, how can I ensure the calling consumer will be in charge of getting that result? I haven't found any filtering capability so far in the docs. I've read https://docs.bullmq.io/guide/returning-job-data#using-a-results-queue but this approach would not enforce the source consumer is the one that will pick the result first.

Thanks!

**Comment by manast** [comment]
If I understand your case correctly, you mean that a given service that is producing a job is expecting to get the result of that job. One way of doing this is by having different "result" queues, one for every service, and having dedicated workers for every result queue. This would be in my opinion the most robust way of achieving what you are asking.
